### PR TITLE
Add integration test for IdentityByState

### PR DESCRIPTION
**Pull request by nitag**
Integration test for IdentityByState (issue 70)

### CODE REVIEW

**Comment by nitag** [comment]
Fixes #70

**Comment by nitag** [comment]
I signed it.

On Tue, Mar 8, 2016 at 10:36 PM, googlebot <notifications@github.com> wrote:

> Thanks for your pull request. It looks like this may be your first
> contribution to a Google open source project. Before we can look at your
> pull request, you'll need to sign a Contributor License Agreement (CLA).
>
> [image: :memo:] *Please visit https://cla.developers.google.com/
> <https://cla.developers.google.com/> to sign.*
>
> Once you've signed, please reply here (e.g. I signed it!) and we'll
> verify. Thanks.
> ------------------------------
>
>    - If you've already signed a CLA, it's possible we don't have your
>    GitHub username or you're using a different email address. Check your
>    existing CLA data <https://cla.developers.google.com/clas> and verify
>    that your email is set on your git commits
>    <https://help.github.com/articles/setting-your-email-in-git/>.
>    - If you signed the CLA as a corporation, please let us know the
>    company's name.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/googlegenomics/dataflow-java/pull/174#issuecomment-194139881>
> .
>



-- 
Regards,
Nita

File: src/test/java/com/google/cloud/genomics/dataflow/pipelines/IdentityByStateITCase.java [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+        "--project=" + helper.getTestProject(),
+        "--runner=BlockingDataflowPipelineRunner",
+        "--stagingLocation=" + helper.getTestStagingGcsFolder(),
+        "--useGrpc=false"
+        };
+    testBase(ARGS);
+  }
+
+  @Test
+  public void testStreamingLocal() throws Exception {
```
Comment (deflaux) [inline src/test/java/com/google/cloud/genomics/dataflow/pipelines/IdentityByStateITCase.java]: `useGrpc` isn't actually an argument to this pipeline, are you able to successfully run this test?

I get error `java.lang.IllegalArgumentException: Class interface com.google.cloud.genomics.dataflow.pipelines.IdentityByState$Options missing a property named 'useGrpc'.` when I run it.

Also I found I needed to do the following when I ran into [this issue](https://github.com/GoogleCloudPlatform/DataflowJavaSDK/issues/122):
(1) grab a [more recent version of alpn](http://mvnrepository.com/artifact/org.mortbay.jetty.alpn/alpn-boot/8.1.7.v20160121) and stick it in `./lib`
(2) update the activation section in pom.xml to use the newer version

File: src/test/java/com/google/cloud/genomics/dataflow/pipelines/IdentityByStateITCase.java [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+
+  @After
+  public void tearDown() throws Exception {
+    for (GcsPath path : helper.gcsUtil.expand(GcsPath.fromUri(outputPrefix + "*"))) {
+      helper.deleteOutput(path.toString());
+    }
+  }
+
+  @Test
+  public void testPaginatedLocal() throws Exception {
```
Comment (deflaux) [inline src/test/java/com/google/cloud/genomics/dataflow/pipelines/IdentityByStateITCase.java]: Please remove this test.  The IdentityByState pipeline only uses gRPC streaming (there is no option to fall back to the paginated request).

File: src/test/java/com/google/cloud/genomics/dataflow/pipelines/IdentityByStateITCase.java [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+        "--variantSetId=" + helper.PLATINUM_GENOMES_DATASET,
+        "--hasNonVariantSegments",
+        "--output=" + outputPrefix,
+        "--useGrpc=false"
+        };
+    testBase(ARGS);
+  }
+
+  @Test
+  public void testPaginatedCloud() throws Exception {
```
Comment (deflaux) [inline src/test/java/com/google/cloud/genomics/dataflow/pipelines/IdentityByStateITCase.java]: Please remove this test.  The IdentityByState pipeline only uses gRPC streaming (there is no option to fall back to the paginated request).

**Comment by deflaux** [comment]
@nitag this is definitely on the right track, thank you!

**Comment by nitag** [comment]
Thanks for the comments. Submitting a new pull request with the merged changes.

**Pull request closed without merging**